crates/cxx-qt-lib/src/types/mod.rs

```diff
@@ -15,6 +15,12 @@ pub use qdatetime::QDateTime;
 mod qhash;
 pub use qhash::{QHash, QHashPair, QHashPair_QString_QVariant};
 
+// Qt 5 has a different QList<T>
+#[cfg(qt_version_major = "5")]
+mod qt5list;
+#[cfg(qt_version_major = "5")]
+pub use qt5list::{Qt5List, Qt5ListElement};
```

Comment:
Yeah you get `error: ‘::QList_i32’ has not been declared; did you mean ‘Qt5List_i32’?` if you try to do `QList_i32` so the developer needs to know that the name is actually `Qt5List`

Reply:
Unless we alias the C++ parts as well :thinking: maybe that'll be enough...

Reply:
I'll try :-)

Reply:
So the Rust and C++ bits can be aliased but then it fails as the CXX `type_id!(...)` doesn't match.

Eg if we put the `type_id!("QVector_bool")` this works with `type QVector_bool = QVector<bool>` but not with `type QList_bool = QList<bool>` (and the same with Qt5List vs QList etc).

So then you need two different implementations to have two different type_id's which ends up the same (?) as without the latest commit. As then you have a QVector and a QList for both Qt 5 and Qt 6 that are almost duplicated code but also have different type_id's.

This might be the best way we can do this for now until we have code generators that can avoid the duplication of every inner type #355 .